src/components/NewMessage/NewMessageTypingIndicator.vue

```diff
@@ -121,6 +127,27 @@ export default {
 		},
 	},
 
+	mounted() {
+		this.observationInterval = setInterval(() => {
+			const currentTime = new Date().getTime()
+			Object.values(this.externalTypingSignals).forEach(([sessionId, timestamp]) => {
+				// If for some reason 'stop' signal wasn't received, check store for expired 'typing' signals
+				// If updated 'typing' signal doesn't come in last 15s, remove it from store
+				if (currentTime - timestamp > 15000) {
+					this.$store.commit('setTyping', {
+						token: this.token,
+						sessionId,
+						typing: false,
+					})
+				}
+			})
+		}, 5000)
+	},
+
+	beforeDestroy() {
+		clearInterval(this.observationInterval)
+	},
```

Comment:
Current solution with setInterval 5s and checking the timestamp if close to the required behavior of 15s indicator lifetime but not equal to.

Stopping time is not connected to the start time, for example
- interval is started on moment `0s`
- `start` is received at `1s` (supposed to be invalidated at `16s`)
- At `15s` interval is checking for invalid typings -> nothing has been found (start was `14s` ago)
- At `30s` interval is checking again -> typing has been found -> removed

Instead of 15 seconds is was removed after 29 seconds.

Reply:
Also, it is not good for performance. Most likely, especially on a small server, most of the time there are no typings. But we have an interval that checks every 5 seconds even if the browser is not active.

Reply:
In any solution, I think all those logic should not be a part of typing indicator UI element. Who is typing is a global state, typings are added in the store, and the store is responsible for inserting indicators and can control when it should be removed after inserting.

While `NewMessageTypingIndicator` is a UI element showing 

With this solution, state will be not correct without rendering UI component, the component's remounting will break this process and it is not reusable.

Reply:
Moved expiration logic to participantsStore. Other comments are based on the new logic